benchmark/csharp/Tpch/VectorizedFunctions.cs

```diff
@@ -0,0 +1,121 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using Apache.Arrow;
+
+namespace Tpch
+{
+    internal static class VectorizedFunctions
+    {
+        internal unsafe static DoubleArray ComputeTotal(DoubleArray price, DoubleArray discount, DoubleArray tax)
+        {
+            if (price.Length != discount.Length || price.Length != tax.Length)
+                throw new ArgumentException("Arrays need to be the same length");
+
+            int length = price.Length;
+            var builder = new ArrowBuffer.Builder<double>(length);
+            Vector<double> ones = new Vector<double>(1d);
+
+            fixed (double* pStartExtendedPrice = price.Values)
+            fixed (double* pStartDiscount = discount.Values)
+            fixed (double* pStartTax = tax.Values)
+            {
+                double* pExtendedPrice = pStartExtendedPrice;
+                double* pDiscount = pStartDiscount;
+                double* pTax = pStartTax;
+                double* pExtendedPriceEnd = pExtendedPrice + length;
+
+                double* pVectorizationEnd = pExtendedPriceEnd - 4;
+
+                while (pExtendedPrice <= pVectorizationEnd)
+                {
+                    Vector<double> exP = Unsafe.ReadUnaligned<Vector<double>>(pExtendedPrice);
+                    Vector<double> dis = Unsafe.ReadUnaligned<Vector<double>>(pDiscount);
+                    Vector<double> t = Unsafe.ReadUnaligned<Vector<double>>(pTax);
+
+                    Vector<double> r = exP * ((ones - dis) * (ones + t));
+
+                    for (int i = 0; i < Vector<double>.Count; i++)
+                    {
+                        builder.Append(r[i]);
+                    }
+
+                    pExtendedPrice += 4;
+                    pDiscount += 4;
+                    pTax += 4;
+                }
+
+                while (pExtendedPrice <= pExtendedPriceEnd)
+                {
+                    builder.Append(*pExtendedPrice * (1 - *pDiscount) * (1 + *pTax));
+
+                    pExtendedPrice++;
```

Comment:
Please use prefix increment operator throughout the files changed: https://github.com/dotnet/spark/blob/master/docs/coding-guidelines/csharp-coding-style.md.